x-pack/filebeat/input/o365audit/auth/auth.go

```diff
@@ -5,38 +5,31 @@
 package auth
 
 import (
-	"fmt"
+	"context"
 
-	"github.com/Azure/go-autorest/autorest/adal"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
 )
 
 // TokenProvider is the interface that wraps an authentication mechanism and
 // allows to obtain tokens.
 type TokenProvider interface {
 	// Token returns a valid OAuth token, or an error.
 	Token() (string, error)
-
-	// Renew must be called to re-authenticate against the oauth2 endpoint if
-	// when the API returns an Authentication error.
-	Renew() error
 }
 
-// servicePrincipalToken extends adal.ServicePrincipalToken with the
+// credentialTokenProvider extends azidentity.ClientSecretCredential with the
 // the TokenProvider interface.
-type servicePrincipalToken adal.ServicePrincipalToken
+type credentialTokenProvider azidentity.ClientSecretCredential
 
 // Token returns an oauth token that can be used for bearer authorization.
-func (provider *servicePrincipalToken) Token() (string, error) {
-	inner := (*adal.ServicePrincipalToken)(provider)
-	if err := inner.EnsureFresh(); err != nil {
-		return "", fmt.Errorf("refreshing spt token: %w", err)
+func (provider *credentialTokenProvider) Token() (string, error) {
+	inner := (*azidentity.ClientSecretCredential)(provider)
+	tk, err := inner.GetToken(
+		context.TODO(), policy.TokenRequestOptions{Scopes: []string{"https://manage.office.com/.default"}},
```

Comment:
Is this a TODO or a Background?

Where is this scope from? I found discussion [here](https://stackoverflow.com/questions/58402344/what-scope-and-or-resource-to-set-to-get-token-that-works-with-office-365-api/58403936#58403936), but could not find official docs or references in the package or elsewhere in beats.

Reply:
I took the scope value from our other use of this API:
https://github.com/elastic/integrations/blob/a74f0b/packages/o365/data_stream/audit/manifest.yml#L56

The other value I saw around is `https://graph.microsoft.com/.default` but I think the [Office 365 Management Activity API](https://learn.microsoft.com/en-us/office/office-365-management-api/office-365-management-activity-api-reference) is not one of the Graph APIs.

I fixed the context propagation.

Reply:
Thanks